packages/fuel-gauge/src/predicate/predicate-arguments.test.ts

```diff
@@ -140,20 +144,19 @@ describe('Predicate', () => {
       const predicate = new Predicate<[Validation]>(
         predicateBytesMainArgsStruct,
         provider,
-        predicateAbiMainArgsStruct
+        predicateAbiMainArgsStruct,
+        [{ has_account: true, total_complete: 100 }]
       );
 
       const initialPredicateBalance = await fundPredicate(wallet, predicate, amountToPredicate);
       const initialReceiverBalance = await receiver.getBalance();
 
       // #region predicate-struct-arg
-      // #context const predicate = new Predicate(bytecode, chainId, abi);
-      const tx = await predicate
-        .setData({
-          has_account: true,
-          total_complete: 100,
-        })
-        .transfer(receiver.address, amountToReceiver, BaseAssetId, { gasPrice, gasLimit: 10_000 });
+      // #context const predicate = new Predicate(bytecode, chainId, abi, [{ has_account: true, total_complete: 100 }]);
```

Comment:
Using context seems unnecessary here, we could just use the actual predicate instantiation and ensure future compatibility.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1826/commits/2b103dc0a5e7264eb84bb3e945024841ee4d78d7